os/net/lwip/src/core/ipv4/dhcp.c

```diff
@@ -694,6 +702,28 @@ void dhcp_cleanup(struct netif *netif)
 	}
 }
 
+#if LWIP_NETIF_HOSTNAME
+void dhcp_hostname(struct netif *netif, char *name)
+{
+	char *name_heap;
+
+	if (netif == NULL || name == NULL) {
+		return;
+	}
+
+	if (netif->hostname != NULL) {
+		kmm_free(netif->hostname);
+		netif->hostname = NULL;
+	}
+
+	name_heap = (char *)kmm_malloc(strlen(name) + 1);
+	memcpy(name_heap, name, strlen(name));
+	name_heap[strlen(name)] = '\0';
+
+	netif_set_hostname(netif, name_heap);
```

Comment:
I think here is only place where calls netif_set_hostname.
If I were you I would use netif->hostname directly without name_heap, If I am right. (of course netif_set_hostname can be deleteted)

Reply:
netif->hostname is not an array but it is char pointer type. It can be changed on run-time, so this malloc/free management should be implemented. Thanks for the comment.